scripts/format_notebooks.py

```diff
@@ -124,6 +124,60 @@ def preprocess_cell(self, cell, resources, cell_index):
         return cell, resources
 
 
+class CloudRunnerPreprocessor(preprocessors.Preprocessor):
+    path_resource_name = "cloud_runner_path"
+    metadata_tag = "CloudRunner"  # tag for added cells so that we can find them easily
+    git_branch = "master"
+
+    colab_import_code = """\
+# install StellarGraph if running on Google Colab
+import sys
+if 'google.colab' in sys.modules:
+  !pip install -q stellargraph[demos]"""
+
+    def _binder_url(self, notebook_path):
+        return f"https://mybinder.org/v2/gh/stellargraph/stellargraph/{self.git_branch}?filepath={notebook_path}"
+
+    def _colab_url(self, notebook_path):
+        return f"https://colab.research.google.com/github/stellargraph/stellargraph/blob/{self.git_branch}/{notebook_path}"
+
+    def _binder_badge(self, notebook_path):
+        return f'<a href="{self._binder_url(notebook_path)}" alt="Open In Binder"><img src="https://mybinder.org/badge_logo.svg"/>'
+
+    def _colab_badge(self, notebook_path):
+        return f'<a href="{self._colab_url(notebook_path)}" alt="Open In Colab"><img src="https://colab.research.google.com/assets/colab-badge.svg"/>'
+
+    def preprocess(self, nb, resources):
+        notebook_path = resources[self.path_resource_name]
+        if not notebook_path.startswith("demos/"):
+            print(f"Notebook file path of {notebook_path} didn't start with demo")
```

Comment:
Is this meant to be an error, or just a warning? Also

```suggestion
            print(f"Notebook file path of {notebook_path} didn't start with 'demos/'")
```

Reply:
This is a warning, have improved slightly.